### PR TITLE
PoS deprecate DisconnectBlocksToHeight

### PR DESCRIPTION
**Pull request by AeonSw4n**

### CODE REVIEW

**Comment by AeonSw4n** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1076?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#1077** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1077?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1076** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1076?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1075** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1075?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1074** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1074?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/pos-hypersync-updates`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1076?utm_source=stack-comment).

File: lib/snapshot.go [hunk 1]
```diff
@@ -519,6 +519,7 @@ func (snap *Snapshot) Stop() {
 	// It's important!!!
 }
 
+// DEPRECATED: This function is no longer used in PoS.
```
Comment (lazynina) [inline lib/snapshot.go:522]: Looks like this is no longer used anywhere, can we just delete this whole function?

File: lib/server.go [hunk 2]
```diff
@@ -674,9 +674,7 @@ func NewServer(
 		}
 		glog.Errorf(CLog(Red, "NewServer: Forcing a rollback to the last snapshot epoch because node was not closed "+
 			"properly last time"))
-		if err := _snapshot.ForceResetToLastSnapshot(_chain); err != nil {
-			return nil, errors.Wrapf(err, "NewServer: Problem in ForceResetToLastSnapshot"), true
-		}
+		return nil, errors.Wrapf(err, "NewServer: Restart required"), true
```
Comment (lazynina) [inline lib/server.go:677]: I actually think we need to return the server here and leave out the error. We expect the caller to be able to perform the restart.

File: lib/blockchain.go [hunk 3]
```diff
@@ -2700,6 +2700,7 @@ func (bc *Blockchain) processBlockPoW(desoBlock *MsgDeSoBlock, verifySignatures
 	return isMainChain, false, nil
 }
 
+// DEPRECATED: This function is no longer used in PoS.
 // DisconnectBlocksToHeight will rollback blocks from the db and blockchain structs until block tip reaches the provided
 // blockHeight parameter.
 func (bc *Blockchain) DisconnectBlocksToHeight(blockHeight uint64, snap *Snapshot) error {
```
Comment (tholonious) [inline lib/blockchain.go:2706]: Will it be safe to delete this function in this PR stack?

**Review by tholonious — APPROVED** [review]

File: lib/blockchain.go (hunk 3 above)
Comment (AeonSw4n) [reply lib/blockchain.go:2706]: Yeah this is just a recovery mechanism for snapshot. Safe to remove.

**Pull request merged**